From 15ad4fe6d3644e1dd260afeb0b0638d2b52bc4f6 Mon Sep 17 00:00:00 2001
From: Padmashree Mandri <padmashree.mandri@intel.com>
Date: Wed, 28 Aug 2024 09:26:25 +0000
Subject: [PATCH] Fix audio crash

This patch adds changes to handle audio service carsh,
when no primary card found.

Tests Done:
USB playback and Record
Camera record

Tracked-On: OAM-124062
Signed-off-by: padmashree mandri <padmashree.mandri@intel.com>
---
 audio/aidl/default/alsa/StreamAlsa.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/audio/aidl/default/alsa/StreamAlsa.cpp b/audio/aidl/default/alsa/StreamAlsa.cpp
index e57d538cf9..9f7fcf465a 100644
--- a/audio/aidl/default/alsa/StreamAlsa.cpp
+++ b/audio/aidl/default/alsa/StreamAlsa.cpp
@@ -80,6 +80,10 @@ StreamAlsa::StreamAlsa(StreamContext* context, const Metadata& metadata, int rea
                     device, const_cast<struct pcm_config*>(&mConfig.value()),
                     true /*require_exact_match*/);
         } else {
+            if (device.card == -1) {
+                LOG(ERROR) << __func__ << "No primary card found";
+                return ::android::NO_INIT;
+            } else
             proxy = alsa::openProxyForAttachedDevice(
                     device, const_cast<struct pcm_config*>(&mConfig.value()), mBufferSizeFrames);
         }
-- 
2.34.1

